### PR TITLE
[UD][OPS-12198] Implemented bank id based account name parsing for na…

### PR DESCRIPTION
**Pull request by Ufuoma-Daniel**
…me matching

### CODE REVIEW

File: app/util/NameParsingUtil.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    "obie-rbs-international-production",
+    "obie-rbs-international-eq-production",
+    "obie-ulster-ni-production",
+    "obie-ulster-ni-bankline-production"
+  )
+
+  @SuppressWarnings(Array("org.wartremover.warts.IterableOps"))
+  def bankIdBasedAccountNameParsing(accountName: String): Seq[String] = {
+    if (accountName.contains('&')) {
+      val jointAccountNamesList = accountName.split('&')
```
Comment (barnesjake) [inline app/util/NameParsingUtil.scala:41]: What happens when there's more than 2 people on an account? Almost definitely not very common, but I just had a google and it is possible

Comment (barnesjake) [reply app/util/NameParsingUtil.scala:41]: Actually ignore me, looks like it just goes into the wildcard case and that covers it

File: app/util/NameParsingUtil.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+object NameParsingUtil {
+
+  val bankIdMatchingVariationList: Seq[String] = Seq(
+    "obie-natwest-production",
```
Comment (barnesjake) [inline app/util/NameParsingUtil.scala:22]: Are we sure these should stay hard coded? I wonder if config may be a nicer place for them (although I'm not fussed on this, if they never change then hard coding is fine)

**Review by barnesjake — APPROVED** [review]

File: app/util/NameParsingUtil.scala (hunk 2 above)
Comment (Ufuoma-Daniel) [reply app/util/NameParsingUtil.scala:22]: Moving them to config is not a bad idea actually, could be a further todo since this is already merged.

**Pull request merged**